fossunited/api/hackathon.py

```diff
@@ -313,3 +312,30 @@ def get_localhost_requests_by_team(
             )
 
     return requests_by_team or None
+
+
+@frappe.whitelist()
+def join_team_via_code(team_code: str, user: str):
+    """
+    Join a team using a team code
+
+    Args:
+        code (str): Team code
+        user (str): User email
+
+    Returns:
+        dict: Team document as a dictionary
+    """
+    try:
+        team = frappe.get_doc("FOSS Hackathon Team", team_code)
+    except frappe.exceptions.DoesNotExistError:
+        frappe.throw("Team not found")
+        return "Invalid Code. Team with this code does not exist."
```

Comment:
personal opinion/nitpicky comment - not sure if the entered team code is displayed to the user because that would be better IMO.